macros/src/types/named.rs

```diff
@@ -102,3 +179,18 @@ fn format_field(
 
     Ok(())
 }
+
+fn has_specialized_impl(ident: &Ident) -> bool {
+    ident == "Arc"
+        || ident == "Box"
+        || ident == "BTreeMap"
+        || ident == "BTreeSet"
+        || ident == "Cell"
+        || ident == "Cow"
+        || ident == "HashMap"
+        || ident == "HashSet"
+        || ident == "Option"
+        || ident == "Rc"
+        || ident == "RefCell"
+        || ident == "Vec"
```

Comment:
This looks really ugly, but I'm at a loss for how we can detect this at compile time and still have a graceful fallback...